sdk/core/core-https/src/accessTokenCache.ts

```diff
@@ -48,11 +48,19 @@ export class ExpiringAccessTokenCache implements AccessTokenCache {
     this.cachedToken = accessToken;
   }
 
-  getCachedToken(): AccessToken | undefined {
-    if (
-      this.cachedToken &&
+  readyToRefresh(): boolean {
+    return (
+      !this.cachedToken ||
       Date.now() + this.tokenRefreshBufferMs >= this.cachedToken.expiresOnTimestamp
-    ) {
+    );
+  }
+
+  isExpired(): boolean {
```

Comment:
might it be confusing to callers that `isExpired` returns `false` when there is no token? I'm on the fence, personally.

Reply:
This is a bit confusing to me too! What about `boolean | null`? or perhaps just true if there's no token?

Reply:
I think making it a tri-state would be more confusing, so maybe `true` on no token, or alternatively flip this into `isTokenValid(): boolean` since obviously not having a token means it's not valid :)

Reply:
ohh I like `isTokenValid`!! Thanks!!